core/src/gossiper.rs

```diff
@@ -0,0 +1,167 @@
+//! Gossiper is actor which is responsible for transaction gossiping
+
+use std::{sync::Arc, time::Duration};
+
+use iroha_config::sumeragi::Configuration;
+use iroha_data_model::transaction::{
+    AcceptedTransaction, Transaction, TransactionLimits, VersionedAcceptedTransaction,
+    VersionedSignedTransaction,
+};
+use iroha_p2p::Broadcast;
+use parity_scale_codec::{Decode, Encode};
+use tokio::sync::mpsc;
+
+use crate::{
+    prelude::WorldStateView, queue::Queue, sumeragi::Sumeragi, IrohaNetwork, NetworkMessage,
+};
+
+/// [`Gossiper`] actor handle.
+#[derive(Clone)]
+pub struct GossiperHandle {
+    message_sender: mpsc::Sender<TransactionGossip>,
+}
+
+impl GossiperHandle {
+    /// Send [`TransactionGossip`] to actor
+    pub async fn gossip(&self, gossip: TransactionGossip) {
+        self.message_sender
+            .send(gossip)
+            .await
+            .expect("Gossiper must handle messages until there is at least one handle to it")
+    }
+}
+
+/// Actor to gossip transactions and receive transaction gossips
```

Comment:
isn't this struct only receiving, `GossiperHandler` is responsible for sending?

Reply:
No, we generate gossips every `gossip_period` and broadcast them. 

And `GossiperHandle` is used to provide recieved gossips from `Network` to `Gossiper`.